### DISCUSSION TITLE
Default connection to TiDB causes periodic GC-related crashes

### DISCUSSION THREAD

**Discussion by emchristiansen** [SeaQL/sea-orm#1589] [General]
Creating a MySQL connection to TiDB using default parameter values can interact poorly with TiDB garbage collection.
See [this TiDB forum thread](https://ask.pingcap.com/t/every-10-minutes-my-in-flight-stale-reads-fail).

It appears that (from TiDB's perspective), connecting creates a transaction that persists for the lifetime of the connection.
When TiDB runs GC and finds live transactions that reference data it wants to delete, it fails with the error described in the post.

A workaround is to set `max_lifetime` to something smaller than the GC window.

Tested on `sea-orm` v0.11.2 and TiDB 6.5.1.

**Comment by tyt2y3** [comment]
We do not officially support TiDB as a backend.
However, seems like the only thing we can do is to expose a `max_lifetime` config?
Can you research on whether it can be done via SQLx?
